core/node/metadata_calculator/src/updater.rs

```diff
@@ -151,6 +151,10 @@ impl TreeUpdater {
                     .proof_generation_dal()
                     .insert_proof_generation_details(l1_batch_number, object_key)
                     .await;
+                storage
+                    .tee_proof_generation_dal()
+                    .insert_tee_proof_generation_details(l1_batch_number)
+                    .await;
```

Comment:
I'm not really aware of TEE lifecycle, but it looks a bit suspicious that we both call `create_tee_verifier_input_producer_job` and `insert_tee_proof_generation_details` here.
A few questions:
1. What details are being inserted? I assume that it's a copy-paste from proof generation dal, but there we provide an object key for each batch. Here we only seem to create a job -- so a name like `insert_tee_proof_generation_job` would probably reflect the nature of action better.
2. Would TEE proof generation be blocked until tee verifier input is produced? If so, wouldn't it be better for the tee verifier input producer to create a job once the input is produced?

Reply:
[Fixed](https://github.com/matter-labs/zksync-era/pull/1993/commits/79e1d07b6ba6e98a0bb3a9adce9ca38829657e6a). ✅

Fun fact - I used to have the implementation you suggested, but later I arbitrarily switched to a different implementation:
https://github.com/matter-labs/zksync-era/blob/daaf6147ca353753d077b1cdafc41d417c0333e1/core/node/tee_verifier_input_producer/src/lib.rs#L257-L266